scr/controllers/controllers.js

```diff
@@ -0,0 +1,183 @@
+const controller = {};
+const {connector} = require('../../config/mysql_db')
+const  multer = require('multer')
+const storage = require('../../config/multer');
+const uploader = multer({storage});
+
+
+controller.list = (req, res) => {
+        let sql = 'SELECT * FROM new_product';
+        connector.query(sql, (err, results) => {
+            if(err) throw err;
+            res.render('products.ejs', {title : "Product list", results })
+            // console.log(results)
+        });
+};
+
+// controller.search = (req, res) => { funcion searh endpoint
+//   connector.query('SELECT * FROM new_product', (err, results) => {
+//       if(err) throw err;
+//       res.json(results)
+//       data = JSON.stringify();
+//       res.send
+//       // console.log(results)
+//   });
+// }
+
+controller.save = (req, res) => {
+  const {body, file} = req;
+
+  if(body && body.productName && body.productPrice && body.productDetail && file) {
+      let url = `http://localhost:3000/images/${file.filename}`;
+      const sql = `INSERT INTO new_product (product_name, product_price, product_detail, product_img) VALUES (?, ?, ?, ?)`;
+      // Asume que 'connection' es tu cliente de base de datos
+      connector.query(sql, [body.productName, body.productPrice, body.productDetail, url], (err, results) => {
+          if(err) {
+              res.status(500).json({ message: 'There was an error inserting data' });
+          } else {
+              res.status(200).json({
+                  message: 'Data added correctly',
+                  product: {
+                      productName: body.productName,
+                      productPrice: body.productPrice,
+                      productDetail: body.productDetail,
+                      productImage: url
+                  }
+              });
+          }
+      });
+  } else {
+      res.status(400).json({ message: 'Missing required data' });
+  }
+};
+
+controller.edit = (req, res) => {
+    const {product_id} = req.params;
+    const query = `SELECT * FROM new_product WHERE product_id = ?`;
+
+    connector.query(query, [product_id], (error, results) => {
+        if (error) {
+            console.log(error);
+            res.status(500).json('An error occurred');
+        } else {
+            res.render('edit.ejs', {title : "Product edit", data : results[0]});
+        }
+    });
+};
+
+controller.update = (req, res) => {
+
+    const {product_id} = req.params;
+    const data = req.body;
+    const file = req.file;
+    let url = `http://localhost:3000/images/${file.filename}`;
```

Comment:
the base url here should not be hardcoded, because if you publish this project under a domain, it will bring errors. Do some research to find out how to get the pathname or the actual domain where the project has been executed